ocrd/ocrd/cli/workspace.py

```diff
@@ -302,20 +331,36 @@ def workspace_cli_bulk_add(ctx, regex, mimetype, page_id, file_id, url, file_grp
             except KeyError:
                 log.error("Cannot guess mimetype from extension '%s' for '%s'. Set --mimetype explicitly" % (file_path.suffix, file_path))
 
+        # Flag to track whether 'url' should be 'src'
+        url_is_src = False
+
         # expand templates
         for param_name in file_dict:
+            if not file_dict[param_name]:
+                if param_name == 'url':
+                    url_is_src = True
+                    continue
+                raise ValueError(f"OcrdFile attribute '{param_name}' unset ({file_dict})")
             for group_name in group_dict:
                 file_dict[param_name] = file_dict[param_name].replace('{{ %s }}' % group_name, group_dict[group_name])
 
-        # copy files
-        if file_dict['url']:
-            urlpath = Path(workspace.directory, file_dict['url'])
-            if not urlpath.exists():
-                log.info("cp '%s' '%s'", file_path, urlpath)
+        # Where to copy from
+        if 'src' in group_dict:
+            srcpath = Path(group_dict['src'])
```

Comment:
```suggestion
        if file_dict['src']:
            srcpath = Path(file_dict['src'])
```

Reply:
This won't work because `src` is not a key in the original definition of `file_dict`, so it won't be set by the loop before this. I am also missing the commit that wired the `source_path` kwarg, apparently I forgot to push it and/or wasn't on a branch when I did it :(